### ISSUE TITLE
[Server Side] Add Support for cross Browser tab state per user

### ISSUE THREAD

**Issue by vertonghenb** [aspnet/Blazor#1208]
First of all the serversideblazor rocks, this is rather an enhancement.

In modern SPA's people tend to use the `localstorage `or cookies to store cross browser tab data for a user. Since we're working on the server I can image we can store this data cross tab via DI.

Currently we can store user data of the client side by creating an `AppState ` class and adding it to DI as a Scoped Service, which is stored per user per tab. However if we could add a class or service which is cross tab we can eliminate the use of localstorage. 

Basically add a class per user per browser tab/connection in DI, is it even possible?

**Comment by SteveSandersonMS** [comment]
You could do that by storing data in ASP.NET server side session storage.

**Comment by vertonghenb** [comment]
Coming from the docs:

> Session isn't supported in SignalR apps because a SignalR Hub may execute independent of an HTTP context. For example, this can occur when a long polling request is held open by a hub beyond the lifetime of the request's HTTP context.

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/app-state?view=aspnetcore-2.1

**Comment by GoranHalvarsson** [comment]
How about JSON Web Token?https://jwt.io/introduction/

**Comment by vertonghenb** [comment]
I did it with a JWT in localstorage and reloading the state from there but it can be made simplier
